homeassistant/components/google/__init__.py

```diff
@@ -97,18 +97,21 @@
 
 
 CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_CLIENT_ID): cv.string,
-                vol.Required(CONF_CLIENT_SECRET): cv.string,
-                vol.Optional(CONF_TRACK_NEW, default=True): cv.boolean,
-                vol.Optional(CONF_CALENDAR_ACCESS, default="read_write"): cv.enum(
-                    FeatureAccess
-                ),
-            }
-        )
-    },
+    vol.All(
+        cv.deprecated(DOMAIN),
```

Comment:
There will be a double warning if we use this and also do the manual warning log below.

Reply:
I noticed this when setting up Spotify and we decided to leave this as the second error message has a bit more context about application credentials:
https://github.com/home-assistant/core/pull/71871#discussion_r873200554

I've since copied this pattern for every application credentials deprecation -- though it would not difficult if we want to change this up.

Reply:
Ok.